extension/src/accessors.rs

```diff
@@ -1,271 +1,120 @@
+use std::convert::TryInto as _;
 
-use counter_agg::range::I64Range;
 use pgx::*;
 
-use flat_serialize::*;
-
-use std::convert::TryInto;
-
-use crate::{
-    build, flatten,
-    pg_type,
-    ron_inout_funcs,
-};
+use counter_agg::range::I64Range;
+use flat_serialize::FlatSerializable;
+
+use crate::{build, flatten, pg_type, ron_inout_funcs};
+
+macro_rules! accessor {
```

Comment:
I _think_ this can be simplified to use

```rust
macro_rules! accessor {
    (
        $name: ident (
            $($field:ident : $typ: tt),* $(,)?
        )
    ) => {
         // ...
    };
}

macro_rules! accessor_fn_impl {
    (
        $name: ident (
            $( $field:ident : $typ: tt ),*
                $(,)?
        )
    ) => {
        // ...
    };
}
```

([playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b5cfef7f574ef7196a200f3d0d3c6a8f))

if you think that makes it simpler.

Reply:
I do, and as I indicated on the call, I didn't understand why `vals` was necessary and had tried to get rid of it.  I pushed a commit attempting your suggestion.  Can you tell what is wrong?  Thanks.